src/sys/prctl.rs

```diff
@@ -151,10 +151,11 @@ pub fn get_name() -> Result<CString> {
 
     let res = unsafe { libc::prctl(libc::PR_GET_NAME, &buf, 0, 0, 0) };
 
-    let len = buf.iter().position(|&c| c == 0).unwrap_or(buf.len());
-    let name = CStr::from_bytes_with_nul(&buf[..=len]).map_err(|_| Errno::EINVAL)?;
-
-    Errno::result(res).map(|_| name.to_owned())
+    Errno::result(res).map(|_| {
+        CStr::from_bytes_until_nul(&buf)
+            .map(CStr::to_owned)
+            .map_err(|_| Errno::EINVAL)
+    }).flatten()
```

Comment:
[`Result::flatten()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.flatten) is a nightly API and thus causing CI failures:

```
error[E0658]: use of unstable library feature 'result_flattening'
   --> src/sys/prctl.rs:158:8
    |
158 |     }).flatten()
    |        ^^^^^^^
    |
    = note: see issue #70142 <https://github.com/rust-lang/rust/issues/70142> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `nix` due to previous error

Exit status: 101
```

What about writing it like this:

```rust
    Errno::result(res).and_then(|_| {
        CStr::from_bytes_until_nul(&buf)
            .map(CStr::to_owned)
            .map_err(|_| Errno::EINVAL)
    })
```